compiler/src/dmd/dtemplate.d

```diff
@@ -5058,135 +5058,6 @@ extern (C++) class TemplateInstance : ScopeDsymbol
-            {
-                auto td = s.isTemplateDeclaration();
-                if (!td)
-                    return 0;
-
-                /* If any of the overloaded template declarations need inference,
-                 * then return true
-                 */
-                if (!td.onemember)
-                    return 0;
-                if (auto td2 = td.onemember.isTemplateDeclaration())
-                {
-                    if (!td2.onemember || !td2.onemember.isFuncDeclaration())
-                        return 0;
-                    if (tiargs.length >= td.parameters.length - (td.isVariadic() ? 1 : 0))
-                        return 0;
-                    return 1;
-                }
-                auto fd = td.onemember.isFuncDeclaration();
-                if (!fd || fd.type.ty != Tfunction)
-                    return 0;
-
-                foreach (tp; *td.parameters)
-                {
-                    if (tp.isTemplateThisParameter())
-                        return 1;
-                }
-
-                /* Determine if the instance arguments, tiargs, are all that is necessary
-                 * to instantiate the template.
-                 */
-                //printf("tp = %p, td.parameters.length = %d, tiargs.length = %d\n", tp, td.parameters.length, tiargs.length);
-                auto tf = fd.type.isTypeFunction();
-                if (tf.parameterList.length)
-                {
-                    auto tp = td.isVariadic();
-                    if (tp && td.parameters.length > 1)
-                        return 1;
-
-                    if (!tp && tiargs.length < td.parameters.length)
-                    {
-                        // Can remain tiargs be filled by default arguments?
-                        foreach (size_t i; tiargs.length .. td.parameters.length)
-                        {
-                            if (!(*td.parameters)[i].hasDefaultArg())
-                                return 1;
-                        }
-                    }
-
-                    foreach (i, fparam; tf.parameterList)
-                    {
-                        // 'auto ref' needs inference.
-                        if (fparam.storageClass & STC.auto_)
-                            return 1;
-                    }
-                }
-
-                if (!flag)
-                {
-                    /* Calculate the need for overload resolution.
-                     * When only one template can match with tiargs, inference is not necessary.
-                     */
-                    dedtypes.setDim(td.parameters.length);
-                    dedtypes.zero();
-                    if (td.semanticRun == PASS.initial)
-                    {
-                        if (td._scope)
-                        {
-                            // Try to fix forward reference. Ungag errors while doing so.
-                            Ungag ungag = td.ungagSpeculative();
```

Comment:
... and its unused in the original, so I assume it is reasonable to ignore this

Reply:
We can't break the test suite though, so you have to add a `cast(void) ungag` or an extra dscanner ignore rule.

Reply:
done.